include/seqan/align/dp_align_simd_helper.h

```diff
@@ -0,0 +1,234 @@
+                                       TSimdString & masks,
+                                       std::vector<size_t> & endsH,
+                                       std::vector<size_t> & endsV)
+{
+    using TStringH = typename Value<TStringSetH>::Type;
+    using TStringV = typename Value<TStringSetV>::Type;
+    using TSimdVector = typename Value<TSimdString>::Type;
+    // check if all sequences have the same length
+    unsigned int numAlignments = LENGTH<TSimdVector>::VALUE;
+    bool allEqualH = true, allEqualV = true;
+    std::vector<size_t> seqLengthsH(numAlignments), seqLengthsV(numAlignments);
+    seqLengthsH[0] = length(stringsH[0]);
+    seqLengthsV[0] = length(stringsV[0]);
+
+    for(unsigned i = 1; i < numAlignments; ++i)
+    {
+        seqLengthsH[i] = length(stringsH[i]);
+        seqLengthsV[i] = length(stringsV[i]);
+        allEqualH &= (seqLengthsH[i] == seqLengthsH[i-1]);
+        allEqualV &= (seqLengthsV[i] == seqLengthsV[i-1]);
+    }
+
+    // if yes, create SIMD representation without doing anything else
+    if(allEqualH && allEqualV)
+    {
+        resize(simdH, seqLengthsH[0]);
+        resize(simdV, seqLengthsV[0]);
+        _createSimdRepresentation(simdH, stringsH, seqLengthsH[0]);
+        _createSimdRepresentation(simdV, stringsV, seqLengthsV[0]);
+        return;
+    }
+
+    // otherwise we have to copy the sequences to be able to add a
+    // padding character before calling _createSimdRepresentation,
+    // because all sequences must have the same length
+    size_t maxH = *std::max_element(seqLengthsH.begin(), seqLengthsH.end());
+    size_t maxV = *std::max_element(seqLengthsV.begin(), seqLengthsV.end());
+
+    // and we have to prepare the bit masks of the DPScoutState
+    resize(masks, maxV, createVector<TSimdVector>(0));
+    resize(masksV, maxV, createVector<TSimdVector>(0));
+    resize(masksH, maxH, createVector<TSimdVector>(0));
+
+    // copy strings and add padding chars
+    StringSet<TStringH> paddedH;
+    StringSet<TStringV> paddedV;
+    resize(paddedH, numAlignments);
+    resize(paddedV, numAlignments);
+
+    for(unsigned i = 0; i < numAlignments; ++i)
+    {
+        // add padding: the padding character should be part of the amino acid alphabet
+        // otherwise a possible score matrix look-up fails, we use 'A' therefor
+        assignValue(paddedH, i, stringsH[i]);
+        assignValue(paddedV, i, stringsV[i]);
+        resize(paddedH[i], maxH, 'A');
+        resize(paddedV[i], maxV, 'A');
+
+        // mark the original end position of the alignment in the masks (with -1, all bits set)
+        assignValue(masksH[seqLengthsH[i]-1], i, -1);
+        assignValue(masksV[seqLengthsV[i]-1], i, -1);
+        endsH.push_back(seqLengthsH[i]-1);
+        endsV.push_back(seqLengthsV[i]-1);
+    }
+
+    // sort the end positions, remove duplicates
+    std::sort(endsH.begin(), endsH.end());
+    endsH.erase( std::unique( endsH.begin(), endsH.end() ), endsH.end() );
```

Comment:
Well the whole function is kind of enhanceable, but I thought, I will adapt this anyway, when improving the dp interfaces. But I agree, I'll better have another look to it.